openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
-    function findAalWeight(pdData, aal) {
-        // Find the aal value
-        if (aal.some(function(currentValue) {
-            return (pdData.name == currentValue);
-        })) {
-            tempAalWeight = pdData.weight;
-        }
-        (pdData.children || []).forEach(function(currentItem) {
-            findAalWeight(currentItem, [aal]);
-        });
-    }
+    // process each Social Vulnerability Index nodes
+    // Get all the primary indicators
+    var ct = 0;
 
-    function utfGridClickEvent(utfGrid) {
-        utfGrid.on('click', function(e) {
-            processIndicators(e);
-            selectedGrid = e;
-        });
+    for (var i = 0; i < socialVulnIndex.length; i++) {
+        for (var e = 0 ; e < socialVulnIndex[i].children.length; e++, ct++ ) {
+            allPrimaryIndicators.push(socialVulnIndex[i].children[e].field);
+        }
     }
 
-    function processIndicators(e) {
-        var districName = e.data.name_1;
-        //$('#pcp-charts').dialog('open');
-        // Get the SVIR data from the utfGrid
-        var tmpIri = {};
-        var tmpPI;
-        var tempCI;
-        if (e.data) {
-            // Find the variables that are primary indicators and their respecive category
-            var ci = 'categoryIndicator';
-            findCategoryIndicators(pdData, [ci]);
-            var pi = 'primaryIndicator';
-            findPrimaryIndicators(pdData, [pi]);
-            var svi = 'svi';
-            findSvi(pdData, [svi]);
-
-            // Create an object for each of the category indicators
-            for (var n = 0; n < pdTempCategoryIndicator.length; n++) {
-                parentChildKey[pdTempCategoryIndicator[n]] = [];
-                var category = pdTempCategoryIndicator[n];
-            }
-            var iri = e.data.ir;
-            iri = iri.split(',');
-            var munic = e.data.municipio;
-            munic = munic.split(',');
-            var pri = e.data.pri;
-
-            if (pri == undefined) {
-                pri = e.data.aal;
-            }
-            var munic_num = e.data['municipio'].split(',').length;
-            municipality = e.data['municipio'].split(',');
-
+    /////////////////////////////////////////
+    //// Compute the Category indicators ////
+    /////////////////////////////////////////
 
-            /////////////////////////////////////////////
-            //// Create the primary indicator objects ///
-            /////////////////////////////////////////////
+    //build the category indicator object
+    var catData = [];
+    tempString = [];
 
-            for (var k in weightedPrimaryIndicator) {
-                delete weightedPrimaryIndicator[k];
+    function generateObject(temp) {
+        var munic = temp[0];
+        var theme = temp[1];
+        var value = parseFloat(temp[2]);
+        // add the theme and value to each category data object
+        for (var i = 0; i < catData.length; i++) {
+            if (catData[i].region == munic) {
```

Comment:
The structure of `catData` might be improved, to keep logically separated the theme values and the region. A possible solution would be to nest theme values one step further. So you would have `catData[i]` containing `region` (or `zone`) as the name of the zone, and `themeValues` as a dictionary theme -> value. In this way, you can iterate through the themeValues without checking every time if you are accessing the zone name.